File: frontend/src/pages/Collaboration.js
import { useState, useEffect } from 'react';
import { useAuth } from '../contexts/AuthContext';
import { useBrand } from '../contexts/BrandContext';
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '../components/ui/card';
import { Button } from '../components/ui/button';
import { Badge } from '../components/ui/badge';
import { Input } from '../components/ui/input';
import { Label } from '../components/ui/label';
import { Textarea } from '../components/ui/textarea';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '../components/ui/select';
import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle, DialogTrigger } from '../components/ui/dialog';
import { Tabs, TabsContent, TabsList, TabsTrigger } from '../components/ui/tabs';
import { toast } from 'sonner';
import {
  Users, Plus, Check, X, RotateCcw, MessageSquare,
  Loader2, Clock, CheckCircle2, XCircle, AlertCircle,
  Activity, Send, Trash2, ChevronDown
} from 'lucide-react';
import axios from 'axios';

const API = `${process.env.REACT_APP_BACKEND_URL}/api`;

const STATUS_CONFIG = {
  pending: { label: 'Pendente', color: 'bg-amber-100 text-amber-700 border-amber-300', icon: Clock },
  approved: { label: 'Aprovado', color: 'bg-green-100 text-green-700 border-green-300', icon: CheckCircle2 },
  rejected: { label: 'Rejeitado', color: 'bg-red-100 text-red-700 border-red-300', icon: XCircle },
  changes_requested: { label: 'Alteracoes', color: 'bg-blue-100 text-blue-700 border-blue-300', icon: RotateCcw }
};

const ITEM_TYPES = [
  { value: 'pillar', label: 'Pilar' },
  { value: 'touchpoint', label: 'Touchpoint' },
  { value: 'strategy', label: 'Estrategia' },
  { value: 'content', label: 'Conteudo' },
  { value: 'report', label: 'Relatorio' }
];

const ACTION_LABELS = {
  approval_created: 'Solicitou aprovacao',
  approval_approve: 'Aprovou',
  approval_reject: 'Rejeitou',
  approval_request_changes: 'Pediu alteracoes',
  comment_added: 'Comentou',
  page_view: 'Visualizou'
};

function timeAgo(dateStr) {
  if (!dateStr) return '';
  const d = new Date(dateStr);
  const diff = Math.floor((Date.now() - d.getTime()) / 1000);
  if (diff < 60) return 'agora';
  if (diff < 3600) return `${Math.floor(diff / 60)}min`;
  if (diff < 86400) return `${Math.floor(diff / 3600)}h`;
  return `${Math.floor(diff / 86400)}d`;
}

export default function Collaboration() {
  const { token, user } = useAuth();
  const { currentBrand } = useBrand();
  const [loading, setLoading] = useState(true);
  const [approvals, setApprovals] = useState([]);
  const [counts, setCounts] = useState({});
  const [activities, setActivities] = useState([]);
  const [comments, setComments] = useState([]);
  const [dialogOpen, setDialogOpen] = useState(false);
  const [selectedApproval, setSelectedApproval] = useState(null);
  const [actionComment, setActionComment] = useState('');
  const [newComment, setNewComment] = useState('');
  const [commentFilter, setCommentFilter] = useState({ type: '', id: '' });
  const [formData, setFormData] = useState({ item_type: 'pillar', item_id: '', item_name: '', description: '' });

  useEffect(() => { if (currentBrand) loadData(); }, [currentBrand]);

  const loadData = async () => {
    setLoading(true);
    try {
      const [appRes, actRes, cmtRes] = await Promise.all([
        axios.get(`${API}/brands/${currentBrand.brand_id}/approvals`, { headers: { Authorization: `Bearer ${token}` } }),
        axios.get(`${API}/brands/${currentBrand.brand_id}/activity?limit=30`, { headers: { Authorization: `Bearer ${token}` } }),
        axios.get(`${API}/brands/${currentBrand.brand_id}/comments`, { headers: { Authorization: `Bearer ${token}` } })
      ]);
      setApprovals(appRes.data.approvals || []);
      setCounts(appRes.data.counts || {});
      setActivities(actRes.data.activities || []);
      setComments(cmtRes.data.comments || []);
    } catch { /* empty */ } finally { setLoading(false); }
  };

  const handleCreateApproval = async () => {
    if (!formData.item_name.trim()) { toast.error('Nome e obrigatorio'); return; }
    try {
      await axios.post(`${API}/brands/${currentBrand.brand_id}/approvals`, formData, { headers: { Authorization: `Bearer ${token}` } });
      toast.success('Solicitacao de aprovacao criada!');
      setDialogOpen(false);
      setFormData({ item_type: 'pillar', item_id: '', item_name: '', description: '' });
      loadData();
    } catch { toast.error('Erro ao criar'); }
  };

  const handleApprovalAction = async (approvalId, action) => {
    try {
      await axios.post(`${API}/brands/${currentBrand.brand_id}/approvals/${approvalId}/action`,
        { action, comment: actionComment },
        { headers: { Authorization: `Bearer ${token}` } }
      );
      const labels = { approve: 'Aprovado!', reject: 'Rejeitado', request_changes: 'Alteracoes solicitadas' };
      toast.success(labels[action]);
      setSelectedApproval(null);
      setActionComment('');
      loadData();
    } catch { toast.error('Erro ao processar'); }
  };

  const handleAddComment = async () => {
    if (!newComment.trim()) return;
    try {
      await axios.post(`${API}/brands/${currentBrand.brand_id}/comments`,
        { item_type: 'general', item_id: 'brand', content: newComment },
        { headers: { Authorization: `Bearer ${token}` } }
      );
      setNewComment('');
      loadData();
    } catch { toast.error('Erro ao comentar'); }
  };

  const handleDeleteComment = async (commentId) => {
    try {
      await axios.delete(`${API}/brands/${currentBrand.brand_id}/comments/${commentId}`, { headers: { Authorization: `Bearer ${token}` } });
      loadData();
    } catch { toast.error('Erro ao remover'); }
  };

  if (!currentBrand) return <div className="flex items-center justify-center h-64"><p className="text-muted-foreground">Selecione uma marca primeiro</p></div>;
  if (loading) return <div className="flex items-center justify-center h-64"><Loader2 className="h-8 w-8 animate-spin text-primary" /></div>;

  return (
    <div className="space-y-6" data-testid="collaboration-page">
      <div className="flex items-center justify-between">
        <div className="flex items-center gap-4">
          <div className="w-12 h-12 rounded-xl bg-gradient-to-br from-blue-500 to-indigo-600 flex items-center justify-center">
            <Users className="h-6 w-6 text-white" />
          </div>
          <div>
            <h1 className="text-2xl font-bold">Colaboracao</h1>
            <p className="text-muted-foreground">Aprovacoes, comentarios e atividades da equipe</p>
          </div>
        </div>
        <Dialog open={dialogOpen} onOpenChange={setDialogOpen}>
          <DialogTrigger asChild>
            <Button data-testid="new-approval-btn"><Plus className="h-4 w-4 mr-2" /> Nova Aprovacao</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader><DialogTitle>Solicitar Aprovacao</DialogTitle></DialogHeader>
            <div className="space-y-4 py-2">
              <div className="space-y-2">
                <Label>Tipo</Label>
                <Select value={formData.item_type} onValueChange={v => setFormData(p => ({ ...p, item_type: v }))}>
                  <SelectTrigger data-testid="approval-type-select"><SelectValue /></SelectTrigger>
                  <SelectContent>{ITEM_TYPES.map(t => <SelectItem key={t.value} value={t.value}>{t.label}</SelectItem>)}</SelectContent>
                </Select>
              </div>
              <div className="space-y-2">
                <Label>Nome do Item *</Label>
                <Input value={formData.item_name} onChange={e => setFormData(p => ({ ...p, item_name: e.target.value }))} placeholder="Ex: Pilar de Proposito" data-testid="approval-name-input" />
              </div>
              <div className="space-y-2">
                <Label>Descricao</Label>
                <Textarea value={formData.description} onChange={e => setFormData(p => ({ ...p, description: e.target.value }))} placeholder="O que precisa ser revisado..." rows={3} />
              </div>
            </div>
            <DialogFooter>
              <Button variant="outline" onClick={() => setDialogOpen(false)}>Cancelar</Button>
              <Button onClick={handleCreateApproval} data-testid="submit-approval-btn">Solicitar</Button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      </div>

      {/* Counts */}
      <div className="grid grid-cols-4 gap-3">
        {Object.entries(STATUS_CONFIG).map(([key, cfg]) => {
          const Icon = cfg.icon;
          return (
            <Card key={key} className={counts[key] > 0 && key === 'pending' ? 'border-amber-300' : ''}>
              <CardContent className="pt-6 text-center">
                <Icon className="h-6 w-6 mx-auto mb-1 text-muted-foreground" />
                <p className="text-2xl font-bold">{counts[key] || 0}</p>
                <p className="text-xs text-muted-foreground">{cfg.label}</p>
              </CardContent>
            </Card>
          );
        })}
      </div>

      <Tabs defaultValue="approvals">
        <TabsList>
          <TabsTrigger value="approvals" data-testid="tab-approvals">Aprovacoes {counts.pending > 0 && <Badge className="ml-1.5 bg-amber-500 text-white text-xs">{counts.pending}</Badge>}</TabsTrigger>
          <TabsTrigger value="comments" data-testid="tab-comments">Comentarios</TabsTrigger>
          <TabsTrigger value="activity" data-testid="tab-activity">Atividades</TabsTrigger>
        </TabsList>

        {/* Approvals */}
        <TabsContent value="approvals" className="space-y-3">
          {approvals.length === 0 ? (
            <Card className="border-dashed"><CardContent className="py-8 text-center text-muted-foreground">Nenhuma solicitacao de aprovacao. Crie uma para iniciar o workflow.</CardContent></Card>
          ) : approvals.map(app => {
            const cfg = STATUS_CONFIG[app.status] || STATUS_CONFIG.pending;
            const Icon = cfg.icon;
            return (
              <Card key={app.approval_id} data-testid={`approval-${app.approval_id}`}>
                <CardContent className="pt-4">
                  <div className="flex items-start justify-between gap-4">
                    <div className="flex-1">
                      <div className="flex items-center gap-2 mb-1">
                        <Badge className={cfg.color}><Icon className="h-3 w-3 mr-1" />{cfg.label}</Badge>
                        <Badge variant="outline" className="text-xs">{ITEM_TYPES.find(t => t.value === app.item_type)?.label || app.item_type}</Badge>
                      </div>
                      <p className="font-semibold">{app.item_name}</p>
                      {app.description && <p className="text-sm text-muted-foreground mt-1">{app.description}</p>}
                      <p className="text-xs text-muted-foreground mt-2">Por {app.requested_by_name} - {timeAgo(app.created_at)}</p>

                      {/* History */}
                      {app.history?.length > 1 && (
                        <div className="mt-3 border-t pt-2 space-y-1">
                          {app.history.slice(1).map((h, i) => (
                            <div key={i} className="flex items-center gap-2 text-xs text-muted-foreground">
                              <span className="font-medium">{h.by_name}</span>
                              <span>{h.action === 'approve' ? 'aprovou' : h.action === 'reject' ? 'rejeitou' : 'pediu alteracoes'}</span>
                              {h.comment && <span className="italic">"{h.comment}"</span>}
                              <span>{timeAgo(h.at)}</span>
                            </div>
                          ))}
                        </div>
                      )}
                    </div>

                    {/* Actions */}
                    {app.status === 'pending' && (
                      <div className="flex flex-col gap-1 shrink-0">
                        {selectedApproval === app.approval_id ? (
                          <div className="space-y-2 w-48">
                            <Input placeholder="Comentario (opcional)" value={actionComment} onChange={e => setActionComment(e.target.value)} className="text-xs" />
                            <div className="flex gap-1">
                              <Button size="sm" className="flex-1 bg-green-600 hover:bg-green-700 h-7 text-xs" onClick={() => handleApprovalAction(app.approval_id, 'approve')}>
                                <Check className="h-3 w-3 mr-1" /> Aprovar
                              </Button>
                              <Button size="sm" variant="destructive" className="flex-1 h-7 text-xs" onClick={() => handleApprovalAction(app.approval_id, 'reject')}>
                                <X className="h-3 w-3 mr-1" /> Rejeitar
                              </Button>
                            </div>
                            <Button size="sm" variant="outline" className="w-full h-7 text-xs" onClick={() => handleApprovalAction(app.approval_id, 'request_changes')}>
                              <RotateCcw className="h-3 w-3 mr-1" /> Pedir Alteracoes
                            </Button>
                            <Button size="sm" variant="ghost" className="w-full h-7 text-xs" onClick={() => setSelectedApproval(null)}>Cancelar</Button>
                          </div>
                        ) : (
                          <Button size="sm" variant="outline" onClick={() => setSelectedApproval(app.approval_id)} data-testid={`review-${app.approval_id}`}>Revisar</Button>
                        )}
                      </div>
                    )}
                  </div>
                </CardContent>
              </Card>
            );
          })}
        </TabsContent>

        {/* Comments */}
        <TabsContent value="comments" className="space-y-4">
          {/* New Comment */}
          <Card>
            <CardContent className="pt-4">
              <div className="flex gap-2">
                <div className="w-8 h-8 rounded-full bg-primary/10 flex items-center justify-center shrink-0">
                  <span className="text-xs font-bold text-primary">{user?.name?.charAt(0) || '?'}</span>
                </div>
                <div className="flex-1 flex gap-2">
                  <Input placeholder="Escreva um comentario..." value={newComment} onChange={e => setNewComment(e.target.value)}
                    onKeyDown={e => e.key === 'Enter' && handleAddComment()} data-testid="comment-input" />
                  <Button size="icon" onClick={handleAddComment} disabled={!newComment.trim()} data-testid="send-comment-btn"><Send className="h-4 w-4" /></Button>
                </div>
              </div>
            </CardContent>
          </Card>

          {comments.length === 0 ? (
            <Card className="border-dashed"><CardContent className="py-8 text-center text-muted-foreground">Nenhum comentario. Inicie a conversa com sua equipe!</CardContent></Card>
          ) : (
            <div className="space-y-2">
              {comments.map(cmt => (
                <Card key={cmt.comment_id} data-testid={`comment-${cmt.comment_id}`}>
                  <CardContent className="pt-3 pb-3">
                    <div className="flex items-start gap-3">
                      <div className="w-8 h-8 rounded-full bg-primary/10 flex items-center justify-center shrink-0">
                        {cmt.user_picture ? <img src={cmt.user_picture} alt="" className="w-8 h-8 rounded-full" /> :
                         <span className="text-xs font-bold text-primary">{cmt.user_name?.charAt(0) || '?'}</span>}
                      </div>
                      <div className="flex-1 min-w-0">
                        <div className="flex items-center gap-2">
                          <span className="font-medium text-sm">{cmt.user_name}</span>
                          <span className="text-xs text-muted-foreground">{timeAgo(cmt.created_at)}</span>
                          {cmt.item_type !== 'general' && <Badge variant="outline" className="text-xs">{cmt.item_type}</Badge>}
                        </div>
                        <p className="text-sm mt-1">{cmt.content}</p>
                      </div>
                      {cmt.user_id === user?.user_id && (
                        <Button variant="ghost" size="icon" className="h-7 w-7 text-red-400" onClick={() => handleDeleteComment(cmt.comment_id)}>
                          <Trash2 className="h-3 w-3" />
                        </Button>
                      )}
                    </div>
                  </CardContent>
                </Card>
              ))}
            </div>
          )}
        </TabsContent>

        {/* Activity Log */}
        <TabsContent value="activity" className="space-y-2">
          {activities.length === 0 ? (
            <Card className="border-dashed"><CardContent className="py-8 text-center text-muted-foreground">Nenhuma atividade registrada ainda.</CardContent></Card>
          ) : (
            <Card>
              <CardHeader><CardTitle className="text-base flex items-center gap-2"><Activity className="h-5 w-5" /> Log de Atividades</CardTitle></CardHeader>
              <CardContent>
                <div className="space-y-3">
                  {activities.map(act => (
                    <div key={act.activity_id} className="flex items-start gap-3 p-2 rounded hover:bg-muted/30" data-testid={`activity-${act.activity_id}`}>
                      <div className="w-2 h-2 rounded-full bg-blue-500 mt-2 shrink-0" />
                      <div className="flex-1 min-w-0">
                        <p className="text-sm"><span className="font-medium">{act.user_name}</span> {ACTION_LABELS[act.action] || act.action}</p>
                        <p className="text-xs text-muted-foreground">{act.description}</p>
                      </div>
                      <span className="text-xs text-muted-foreground shrink-0">{timeAgo(act.created_at)}</span>
                    </div>
                  ))}
                </div>
              </CardContent>
            </Card>
          )}
        </TabsContent>
      </Tabs>
    </div>
  );
}

File: backend/routes/collaboration.py
"""Collaboration Module - Approvals, Activity Log, Comments"""
from fastapi import APIRouter, HTTPException, Depends
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime, timezone
import uuid

from config import db
from utils.helpers import get_current_user
from routes.notifications import create_notification

router = APIRouter(tags=["Collaboration"])


class ApprovalRequest(BaseModel):
    item_type: str  # pillar, touchpoint, strategy, content
    item_id: str
    item_name: str
    description: Optional[str] = ""
    assigned_to: Optional[str] = None  # user_id of reviewer

class ApprovalAction(BaseModel):
    action: str  # approve, reject, request_changes
    comment: Optional[str] = ""

class CommentRequest(BaseModel):
    item_type: str
    item_id: str
    content: str
    parent_id: Optional[str] = None


# ---- APPROVALS ----

@router.get("/brands/{brand_id}/approvals")
async def get_approvals(brand_id: str, status: str = None, user: dict = Depends(get_current_user)):
    query = {"brand_id": brand_id}
    if status:
        query["status"] = status
    approvals = await db.approvals.find(query, {"_id": 0}).sort("created_at", -1).to_list(100)

    counts = {
        "pending": await db.approvals.count_documents({"brand_id": brand_id, "status": "pending"}),
        "approved": await db.approvals.count_documents({"brand_id": brand_id, "status": "approved"}),
        "rejected": await db.approvals.count_documents({"brand_id": brand_id, "status": "rejected"}),
        "changes_requested": await db.approvals.count_documents({"brand_id": brand_id, "status": "changes_requested"})
    }

    return {"approvals": approvals, "counts": counts}


@router.post("/brands/{brand_id}/approvals")
async def create_approval(brand_id: str, data: ApprovalRequest, user: dict = Depends(get_current_user)):
    approval_id = f"appr_{uuid.uuid4().hex[:12]}"
    doc = {
        "approval_id": approval_id, "brand_id": brand_id,
        "item_type": data.item_type, "item_id": data.item_id, "item_name": data.item_name,
        "description": data.description, "status": "pending",
        "requested_by": user["user_id"], "requested_by_name": user.get("name", ""),
        "assigned_to": data.assigned_to,
        "history": [{"action": "created", "by": user["user_id"], "by_name": user.get("name", ""), "at": datetime.now(timezone.utc).isoformat()}],
        "created_at": datetime.now(timezone.utc).isoformat(), "updated_at": datetime.now(timezone.utc).isoformat()
    }
    await db.approvals.insert_one(doc)
    doc.pop("_id", None)

    # Log activity
    await log_activity(brand_id, user, "approval_created", f"Solicitou aprovacao: {data.item_name}", {"approval_id": approval_id, "item_type": data.item_type})

    # Notify assigned reviewer
    if data.assigned_to:
        await create_notification(
            user_id=data.assigned_to, brand_id=brand_id,
            notif_type="approval_request",
            title="Nova solicitacao de aprovacao",
            message=f"{user.get('name', 'Alguem')} solicitou sua aprovacao para: {data.item_name}",
            link=f"/collaboration",
            actor_name=user.get("name", "")
        )

    return doc


@router.post("/brands/{brand_id}/approvals/{approval_id}/action")
async def approval_action(brand_id: str, approval_id: str, data: ApprovalAction, user: dict = Depends(get_current_user)):
    approval = await db.approvals.find_one({"approval_id": approval_id, "brand_id": brand_id})
    if not approval:
        raise HTTPException(status_code=404, detail="Aprovacao nao encontrada")

    status_map = {"approve": "approved", "reject": "rejected", "request_changes": "changes_requested"}
    new_status = status_map.get(data.action)
    if not new_status:
        raise HTTPException(status_code=400, detail="Acao invalida")

    history_entry = {
        "action": data.action, "comment": data.comment,
        "by": user["user_id"], "by_name": user.get("name", ""),
        "at": datetime.now(timezone.utc).isoformat()
    }

    await db.approvals.update_one(
        {"approval_id": approval_id},
        {"$set": {"status": new_status, "updated_at": datetime.now(timezone.utc).isoformat()}, "$push": {"history": history_entry}}
    )

    action_labels = {"approve": "Aprovou", "reject": "Rejeitou", "request_changes": "Pediu alteracoes"}
    await log_activity(brand_id, user, f"approval_{data.action}", f"{action_labels[data.action]}: {approval.get('item_name', '')}", {"approval_id": approval_id})

    # Notify the original requester about the action
    requester_id = approval.get("requested_by")
    if requester_id and requester_id != user["user_id"]:
        status_labels = {"approve": "aprovada", "reject": "rejeitada", "request_changes": "com alteracoes solicitadas"}
        await create_notification(
            user_id=requester_id, brand_id=brand_id,
            notif_type="approval_action",
            title=f"Aprovacao {status_labels.get(data.action, 'atualizada')}",
            message=f"{user.get('name', 'Alguem')} {action_labels[data.action].lower()}: {approval.get('item_name', '')}",
            link=f"/collaboration",
            actor_name=user.get("name", "")
        )

    updated = await db.approvals.find_one({"approval_id": approval_id}, {"_id": 0})
    return updated


# ---- ACTIVITY LOG ----

async def log_activity(brand_id: str, user: dict, action: str, description: str, metadata: dict = None):
    doc = {
        "activity_id": f"act_{uuid.uuid4().hex[:12]}",
        "brand_id": brand_id,
        "user_id": user["user_id"],
        "user_name": user.get("name", ""),
        "action": action,
        "description": description,
        "metadata": metadata or {},
        "created_at": datetime.now(timezone.utc).isoformat()
    }
    await db.activity_log.insert_one(doc)


@router.get("/brands/{brand_id}/activity")
async def get_activity_log(brand_id: str, limit: int = 50, user: dict = Depends(get_current_user)):
    activities = await db.activity_log.find({"brand_id": brand_id}, {"_id": 0}).sort("created_at", -1).to_list(limit)
    return {"activities": activities, "total": len(activities)}


@router.post("/brands/{brand_id}/activity/log")
async def log_user_activity(brand_id: str, data: dict, user: dict = Depends(get_current_user)):
    """Endpoint for frontend to log user actions."""
    await log_activity(brand_id, user, data.get("action", "page_view"), data.get("description", ""), data.get("metadata", {}))
    return {"success": True}


# ---- COMMENTS ----

@router.get("/brands/{brand_id}/comments")
async def get_comments(brand_id: str, item_type: str = None, item_id: str = None, user: dict = Depends(get_current_user)):
    query = {"brand_id": brand_id}
    if item_type:
        query["item_type"] = item_type
    if item_id:
        query["item_id"] = item_id

    comments = await db.comments.find(query, {"_id": 0}).sort("created_at", -1).to_list(100)
    return {"comments": comments, "total": len(comments)}


@router.post("/brands/{brand_id}/comments")
async def create_comment(brand_id: str, data: CommentRequest, user: dict = Depends(get_current_user)):
    comment_id = f"cmt_{uuid.uuid4().hex[:12]}"
    doc = {
        "comment_id": comment_id, "brand_id": brand_id,
        "item_type": data.item_type, "item_id": data.item_id,
        "content": data.content, "parent_id": data.parent_id,
        "user_id": user["user_id"], "user_name": user.get("name", ""),
        "user_picture": user.get("picture", ""),
        "created_at": datetime.now(timezone.utc).isoformat()
    }
    await db.comments.insert_one(doc)
    doc.pop("_id", None)

    await log_activity(brand_id, user, "comment_added", f"Comentou em {data.item_type}", {"comment_id": comment_id, "item_type": data.item_type, "item_id": data.item_id})

    # Notify other collaborators on the same item (except commenter)
    other_commenters = await db.comments.distinct("user_id", {
        "brand_id": brand_id, "item_type": data.item_type, "item_id": data.item_id,
        "user_id": {"$ne": user["user_id"]}
    })
    for uid in other_commenters:
        await create_notification(
            user_id=uid, brand_id=brand_id,
            notif_type="comment",
            title="Novo comentario",
            message=f"{user.get('name', 'Alguem')} comentou em {data.item_type}: {data.content[:80]}",
            link=f"/collaboration",
            actor_name=user.get("name", "")
        )

    return doc


@router.delete("/brands/{brand_id}/comments/{comment_id}")
async def delete_comment(brand_id: str, comment_id: str, user: dict = Depends(get_current_user)):
    result = await db.comments.delete_one({"comment_id": comment_id, "brand_id": brand_id, "user_id": user["user_id"]})
    if result.deleted_count == 0:
        raise HTTPException(status_code=404, detail="Comentario nao encontrado ou sem permissao")
    return {"message": "Comentario removido"}
